packages/create/templates/plugin-js/web/plugin.jsx

```diff
@@ -0,0 +1,60 @@
+const { useContext, useEffect, useState/*/ [subscribable] /*/, useCallback, useSyncExternalStore/*/ [] /*/ } = require("react");
+// const { } = require("antd");
+
+const {
+	BaseWebPlugin, PageLayout, Control, ControlContext, notifyErrorHandler,
+} = require("@clusterio/web_ui");
+
+const lib = require("@clusterio/lib");
+const { PluginExampleEvent, PluginExampleRequest/*/ [subscribable] /*/, ExampleSubscribableUpdate/*/ [] /*/ } = require("../messages");
+
+function MyTemplatePage() {
+	let control = useContext(ControlContext);// [subscribable] //
+	const plugin = control.plugins.get("// plugin_name //");
+	const [subscribableData, synced] = plugin.useSubscribableData();// [] //
+
+	return <PageLayout nav={[{ name: "// plugin_name //" }]}>
+		<h2>// plugin_name //</h2>// [subscribable] //
+		Synced: {String(synced)} Data: {JSON.stringify([...subscribableData.values()])}// [] //
+	</PageLayout>;
+}
+
+class WebPlugin extends BaseWebPlugin {// [subscribable] //
+	subscribableData = new lib.EventSubscriber(ExampleSubscribableUpdate, this.control);
+	// [] //
+	async init() {
+		this.pages = [
+			{
+				path: "/// plugin_name //",
+				sidebarName: "// plugin_name //",
+				permission: "// plugin_name //.page.view",
```

Comment:
What is the ask here? All existing plugins contain this field and I don't see any additional code in them for checking permissions.

Reply:
If you look into subspace storage's it uses the permission `subspace_storage.storage.view` both for the page and the GetStorageRequest. It's similar for inventory sync. The permission to access the page is also the permission gating access to the resource the page shows.

What I'm asking for is not to use a permission that looks like "view page" as an example because it sets a bad precedent when making a plugin page. It would be better to set the permission to `undefined` here and have the page always show than to use a view page permission that could be bypassed on the client.

Reply:
Ah I see, yeah that makes sense now. I have updated it with a comment as well as two different values for the permission to use either undefined or the subscription permission.